python/lib/communication/dmod/communication/client.py

```diff
@@ -252,147 +429,243 @@ def _check_if_new_session_needed(self, use_current_values: bool = True) -> bool:
+
+        # If we need to reload, also clear any old session data, but this time proceed with the rest of the function
+        if self.force_reload:
+            self._session_id, self._session_secret, self._session_created = None, None, None
+            # Once we force clearing these to ensure a reload is attempted, reset the attribute
+            self.force_reload = False
+        # If not set to force a reload, we may already have valid session attributes; short here if so
+        elif all([self._session_id, self._session_secret, self._session_created]):
             return False
+
         # If there is a cached session file, we will try to load from it
         if self._cached_session_file.exists():
             try:
-                session_id, secret, created = self.parse_session_auth_text(self._cached_session_file.read_text())
+                session_id, secret, created = self._parse_auth_data(self._cached_session_file.read_text())
                 # Only set if all three read properties are valid
                 if all([session_id, secret, created]):
                     self._session_id = session_id
                     self._session_secret = secret
                     self._session_created = created
+                    self._is_new_session = False
             except Exception as e:
                 pass
             # Return opposite of whether session properties are now set correctly (that would mean don't need a session)
             return not all([self._session_id, self._session_secret, self._session_created])
-        # Otherwise (i.e., don't/can't use current session details + no cached file to load), need a new session
         else:
             return True
 
-    # TODO: ...
-    async def authenticate(self, cached_session_file: Optional[Path] = None):
-        #async with websockets.connect(self.endpoint_uri, ssl=self.client_ssl_context) as websocket:
-        #async with websockets.connect(self.maas_endpoint_uri) as websocket:
-        # return await EditView._authenticate_over_websocket(websocket)
-        # Right now, it doesn't matter as long as it is valid
-        # TODO: Fix this to not be ... fixed ...
-        json_as_dict = {'username': 'someone', 'user_secret': 'something'}
-        response_txt = await self.async_send(data=json.dumps(json_as_dict), await_response=True)
-        try:
-            if cached_session_file is not None and not cached_session_file.is_dir() \
-                    and cached_session_file.parent.is_dir():
-                cached_session_file.write_text(response_txt)
-        except Exception as e:
-            # TODO: consider logging something here, but for now just handle so a bad save file doesn't tank us
-            pass
-        #print('*************** Auth response: ' + json.dumps(response_txt))
-        return self.parse_session_auth_text(response_txt)
-
     @property
-    def is_new_session(self):
-        return self._is_new_session
+    def force_reload(self) -> bool:
```

Comment:
This looks like it does not need to be reimplemented in the subclass. Am I missing something?

Reply:
Not sure what you mean.  This is being introduced by this class (_CachedAuthClient_).  There is no previous declaration/definition.